router.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+    createRootRoute,
+    createRoute,
+    createRouter,
+    Outlet,
+} from '@tanstack/react-router';
+import DashboardPage from './src/pages/DashboardPage';
+import StrategyBuilder from './src/components/StrategyBuilder/StrategyBuilder';
+
+const rootRoute = createRootRoute({
+    component: () => <div><Outlet /></div>,
+});
+
+const homeRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/',
+    component: DashboardPage,
+});
+
+const strategyRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/strategy',
+    component: StrategyBuilder,
+    validateSearch: (search: Record<string, unknown>) => {
+        return {
+            name: typeof search.name === 'string' ? search.name : '',
+        };
+    },
+});
+
+const routeTree = rootRoute.addChildren([homeRoute, strategyRoute]);
+
+export const router = createRouter({ routeTree });
+
+declare module '@tanstack/react-router' {
+    interface Register {
+        router: typeof router;
+    }
+}
```

Comment:
I think this app would be easier to understand using the file router. You might need move some files around but it would make it easier for you to come back to this project later and be able to maintain it.